src/Compilers/CSharp/Portable/Symbols/ConstraintsHelper.cs

```diff
@@ -1310,19 +1312,20 @@ private static bool SatisfiesConstructorConstraint(TypeSymbol typeArgument)
         }
 
         /// <summary>
-        /// Return true if the class type has a public parameterless constructor.
+        /// Return true if the type has a public parameterless constructor;
+        /// false if the parameterless constructor is not public; and
+        /// null if no parameterless constructor.
         /// </summary>
-        private static bool HasPublicParameterlessConstructor(NamedTypeSymbol type)
+        private static bool? HasPublicParameterlessConstructor(NamedTypeSymbol type)
         {
-            Debug.Assert(type.TypeKind == TypeKind.Class);
```

Comment:
Instead of deleting the assertion, should it be updated to `type.TypeKind is TypeKind.Class or TypeKind.Struct`?